airflow/providers/amazon/aws/hooks/redshift.py

```diff
@@ -15,10 +15,19 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-"""Interact with AWS Redshift, using the boto3 library."""
+"""Interact with AWS Redshift, using the boto3 or redshift_connector library."""
 
-from typing import List, Optional
+from typing import Dict, List, Optional, Union
 
+try:
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
```

Comment:
```suggestion
from airflow.compat.functools import cached_property
```

Reply:
fixed in ea1f49073f545e3c0d3eaa0733dc94a1ac206633